crates/rustc_codegen_spirv/src/linker/inline.rs

```diff
@@ -35,15 +35,29 @@ pub fn inline(sess: &Session, module: &mut Module) -> super::Result<()> {
     // Drop all the functions we'll be inlining. (This also means we won't waste time processing
     // inlines in functions that will get inlined)
     let mut dropped_ids = FxHashSet::default();
+    let mut inlined_dont_inlines = Vec::new();
     module.functions.retain(|f| {
         if should_inline(&disallowed_argument_types, &disallowed_return_types, f) {
+            if has_dont_inline(f) {
+                inlined_dont_inlines.push(f.def_id().unwrap());
+            }
             // TODO: We should insert all defined IDs in this function.
             dropped_ids.insert(f.def_id().unwrap());
             false
         } else {
             true
         }
     });
+    if !inlined_dont_inlines.is_empty() {
+        let names = get_names(module);
+        for f in inlined_dont_inlines {
+            sess.warn(&format!(
+                    "function `{}` has `dont_inline` attribute, but need to be inlined because it has illegal argument or return types",
+                    get_name(&names, f)
+                ));
+        }
```

Comment:
```suggestion
        for f in inlined_dont_inlines {
            sess.warn(&format!(
                "`#[inline(never)]` function `{}` needs to be inlined \
                 because it has illegal argument or return types",
                get_name(&names, f)
            ));
        }
```

(I'm not too fond of "illegal" but I can't think of a good substitute, maybe "unsupported" but that doesn't really have the "needs legalization" feel to it so maybe "illegal" is the right choice)